src/QsCompiler/Compiler/Templates/CppInterop.cs

```diff
@@ -0,0 +1,193 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Quantum.QsCompiler.BondSchemas.EntryPoint;
+
+namespace Microsoft.Quantum.QsCompiler.Templates
+{
+    public class ArgumentCpp : Argument
+    {
+        public ArgumentCpp(Argument argument)
+        {
+            this.Name = argument.Name;
+            this.Position = argument.Position;
+            this.Type = argument.Type;
+            this.ArrayType = argument.ArrayType;
+            this.Values = argument.Values.ToList();
+        }
+
+        public string CppType()
+        {
+            return this.Type switch
+            {
+                DataType.BoolType => "char",
+                DataType.IntegerType => "int64_t",
+                DataType.DoubleType => "double",
+                DataType.PauliType => "char",
+                DataType.RangeType => "InteropRange*",
+                DataType.ResultType => "char",
+                DataType.StringType => "const char*",
+                DataType.ArrayType => "InteropArray *",
+                _ => throw new NotSupportedException($"Unsupported argument type {this.Type}")
+            };
+        }
+
+        public string CliOptionString()
+        {
+            if (this.Name.Length == 1)
+            {
+                return $"-{this.Name}";
+            }
+            else
+            {
+                return $"--{this.Name}";
+            }
+        }
+
+        public string CliTypeDescription()
+        {
+            return this.Type switch
+            {
+                DataType.BoolType => $"bool",
+                DataType.IntegerType => $"integer",
+                DataType.DoubleType => $"double",
+                DataType.PauliType => $"Pauli",
+                DataType.RangeType => $"Range (start, step, end)",
+                DataType.ResultType => $"Result",
+                DataType.StringType => $"String",
+                DataType.ArrayType => this.ArrayType switch
+                {
+                    DataType.BoolType => $"bool array",
+                    DataType.IntegerType => $"integer array",
+                    DataType.DoubleType => $"double array",
+                    DataType.PauliType => $"Pauli array",
+                    DataType.RangeType => $"Range array",
+                    DataType.ResultType => $"Result array",
+                    DataType.StringType => $"String array",
+                    _ => throw new NotSupportedException($"Unsupported array type {this.Type}")
+                },
+                _ => throw new NotSupportedException($"Unsupported argument type {this.Type}")
+            };
+        }
+
+        public string CliDescription() => $"A {this.CliTypeDescription()} value for the {this.Name} argument";
+
+        public string? CppVarType()
```

Comment:
I think this method can return a non nullable `string`.

Reply:
done